cirq/testing/circuit_compare.py

```diff
@@ -305,6 +308,75 @@ def assert_has_consistent_apply_unitary(
                                    atol=atol)
 
 
+def _assert_apply_unitary_works_when_axes_transposed(
```

Comment:
This is a nice addition to the battery. Could you add tests for this assert? We do this for a number of other existing ones.

Reply:
Added a failure triggering test.